### PR TITLE
RCQ wakup actively when resource group is in trickle mode

### LINKED ISSUE #8391: tiflash RU usage can is not stable
## Bug Report

Please answer these questions before submitting your issue. Thanks!
Running the ossinsight workload for 2 hours, we can see the RU usage is not stable, sometimes it drop and sometimes it bump
NTwZ2WoRfe### 1. Minimal reproduce step (Required)

### PR DESCRIPTION
**Pull request by guo-shaoge**
### What problem does this PR solve?

Issue Number: close #8391

Problem Summary:
To make RU usage more accurate and stable:
1. add `estWaitDuraMS`, so RCQ can wakeup when token refill in trickle mode

Before:
image

After:
img_v3_025e_73d5e067-9b4c-4398-93d5-9faf242718fg

others:
1. Update where metrics changes occur, rather than uniformly updating it every 5 seconds within `collectMetrics()`, which ensures more accurate metrics.
6. Delete `fetch_tokens_from_gac_count` metrics because it doesn't help debug

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)

1. create resource group
```
create user u5 identified by '1234';
GRANT SELECT,INSERT,UPDATE,DELETE,CREATE,DROP,PROCESS,REFERENCES,ALTER,SHOW DATABASES,SUPER,EXECUTE,INDEX,CREATE USER,CREATE TABLESPACE,TRIGGER,CREATE VIEW,SHOW       VIEW,CREATE ROLE,DROP ROLE,CREATE TEMPORARY TABLES,LOCK TABLES,CREATE ROUTINE,       ALTER ROUTINE,EVENT,RELOAD,FILE,REPLICATION CLIENT,REPLICATION SLAVE ON *.* TO 'u5';
create resource group rg5 ru_per_sec = 80000;
alter user u5 resource group rg5;
```
2. run ossinght workload(events-total) for x hours using 8 concurrency
```
SELECT     COUNT(1) AS cnt,     MAX(created_at) AS latest_created_at,     UNIX_TIMESTAMP(MAX(created_at)) AS latest_timestamp FROM github_events;
```

- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by SeaRise — APPROVED** [review]

File: dbms/src/Flash/ResourceControl/LocalAdmissionController.cpp [hunk 1]
```diff
@@ -180,28 +173,22 @@ std::optional<LocalAdmissionController::AcquireTokenInfo> LocalAdmissionControll
...
+        if (is_periodically_fetch && !resource_group->needFetchToken(now, DEFAULT_FETCH_GAC_INTERVAL))
             return;
 
-        // During trickle mode, no need to fetch tokens from GAC.
-        if (resource_group->inTrickleModeLease(now))
+        // To avoid low token fetch too freqeuent.
+        if (!is_periodically_fetch
+            && !resource_group->needFetchToken(
+                now,
+                std::chrono::duration_cast<std::chrono::seconds>(DEFAULT_LOW_TOKEN_FETCH_INTERVAL)))
```
Comment (windtalker) [inline dbms/src/Flash/ResourceControl/LocalAdmissionController.cpp:184]: Maybe above two `if` can be merged into one as define `needFetchToken` as `needFetchToken(const SteadyClock::time_point & now, bool is_periodically_fetch)` and choose to use `DEFAULT_LOW_TOKEN_FETCH_INTERVAL` or `DEFAULT_FETCH_GAC_INTERVAL` inside `needFetchToken` based on `is_periodically_fetch`?

Comment (guo-shaoge) [reply dbms/src/Flash/ResourceControl/LocalAdmissionController.cpp]: deleted. maybe no big deal.

File: dbms/src/Flash/ResourceControl/TokenBucket.h [hunk 2]
```diff
@@ -95,6 +95,19 @@ class TokenBucket final
         return fmt::format("tokens: {}, fill_rate: {}, capacity: {}", tokens, fill_rate, capacity);
     }
 
+    uint64_t estWaitDuraMS(uint64_t max_wait_dura_ms) const
+    {
+        const auto tokens = peek();
+        if (tokens > 0 || fill_rate_ms == 0.0)
```
Comment (windtalker) [inline dbms/src/Flash/ResourceControl/TokenBucket.h:101]:
```suggestion
        if (tokens >= 0 || fill_rate_ms == 0.0)
```

File: dbms/src/Flash/ResourceControl/TokenBucket.h [hunk 3]
```diff
@@ -95,6 +95,19 @@ class TokenBucket final
...
+            return 100;
+
+        auto est_dura_ms = static_cast<uint64_t>(std::ceil(-tokens / fill_rate_ms)) + 100;
+        if (est_dura_ms > max_wait_dura_ms)
+            est_dura_ms = max_wait_dura_ms;
+
+        return est_dura_ms;
```
Comment (windtalker) [inline dbms/src/Flash/ResourceControl/TokenBucket.h:108]:
```suggestion
        return std::min(est_dura_ms, max_wait_dura_ms);
```

File: dbms/src/Flash/ResourceControl/TokenBucket.h [hunk 4]
```diff
@@ -95,6 +95,19 @@ class TokenBucket final
         return fmt::format("tokens: {}, fill_rate: {}, capacity: {}", tokens, fill_rate, capacity);
     }
 
+    uint64_t estWaitDuraMS(uint64_t max_wait_dura_ms) const
+    {
+        const auto tokens = peek();
+        if (tokens > 0 || fill_rate_ms == 0.0)
+            return 100;
```
Comment (windtalker) [inline dbms/src/Flash/ResourceControl/TokenBucket.h:102]: Don't need check if `100 > max_wait_dura_ms`?

Comment (guo-shaoge) [reply dbms/src/Flash/ResourceControl/TokenBucket.h]: assert maybe enough

File: dbms/src/Flash/ResourceControl/TokenBucket.h (hunk 3 above)
Comment (guo-shaoge) [reply dbms/src/Flash/ResourceControl/TokenBucket.h]: done

File: dbms/src/Flash/ResourceControl/TokenBucket.h (hunk 2 above)
Comment (guo-shaoge) [reply dbms/src/Flash/ResourceControl/TokenBucket.h]: If `tokens` is zero, return value will be `100`. But still changed, maybe it's more clear

**Review by windtalker — APPROVED** [review]
LGTM

**Pull request merged**